winch/codegen/src/visitor.rs

```diff
@@ -1428,7 +1429,10 @@ where
 
         match heap_type {
             WasmHeapType::Func => match style {
-                TableStyle::CallerChecksSignature => self.emit_lazy_init_funcref(table_index),
+                TableStyle::CallerChecksSignature { lazy_init: true } => {
+                    self.emit_lazy_init_funcref(table_index)
+                }
+                _ => unimplemented!("Support for eager table init"),
```

Comment:
If this remains (which I think is fine) mind adding a check in `Config::validate` that returns an error if winch is enabled but lazy init is disabled?